File: ios/Barem/Barem/Engine/VdLookupRepository.swift
import Foundation

final class VdLookupRepository {
    static let shared = VdLookupRepository()

    private let data: VdIlIlceData

    private init() {
        let url = Bundle.main.url(forResource: "vd_il_ilce", withExtension: "json", subdirectory: "memur")
            ?? Bundle.main.url(forResource: "vd_il_ilce", withExtension: "json")
        guard let url, let raw = try? Data(contentsOf: url) else {
            fatalError("Missing memur/vd_il_ilce.json")
        }
        data = try! JSONDecoder().decode(VdIlIlceData.self, from: raw)
    }

    func iller() -> [String] { data.iller }

    func ilceler(il: String) -> [String] { data.ilcelerByIl[il] ?? [] }

    func bolgeselKod(il: String, ilce: String) -> String? {
        data.bolgeselKodByKey["\(il.trimmingCharacters(in: .whitespaces))-\(ilce.trimmingCharacters(in: .whitespaces))"] ?? nil
    }
}

File: app/src/main/java/tr/erdaldemir/barem/data/VdLookupRepository.kt
package tr.erdaldemir.barem.data

import android.content.Context
import com.google.gson.Gson
import tr.erdaldemir.barem.domain.model.VdIlIlceData

/** A!B13→B14→B16 — VD!AG:AH il/ilçe → bölgesel kod (export_vd_il_ilce.py). */
class VdLookupRepository(context: Context) {

    private val data: VdIlIlceData by lazy {
        val json = context.assets.open("memur/vd_il_ilce.json").bufferedReader().use { it.readText() }
        Gson().fromJson(json, VdIlIlceData::class.java)
    }

    fun iller(): List<String> = data.iller

    fun ilceler(il: String): List<String> = data.ilcelerByIl[il].orEmpty()

    /** Excel A!B16 — VD sütun AH; yoksa null (bölgesel tazminat uygulanmaz). */
    fun bolgeselKod(il: String, ilce: String): String? {
        val key = "${il.trim()}-${ilce.trim()}"
        return data.bolgeselKodByKey[key]
    }
}
